crates/cxx-qt-build/src/lib.rs

```diff
@@ -496,6 +505,139 @@ impl CxxQtBuilder {
+                let qrc_name = file_stem.replace('-', "_");
+                // Note this cannot be inside a namespace
+                format!("  Q_INIT_RESOURCE({qrc_name};")
+            })
+            .collect::<Vec<String>>()
+            .join("\n");
+
+        if !init_resources.is_empty() {
+            self.inits.push(CxxQtBuilderInit::new(
+                "cxx_qt_init_resources",
+                &formatdoc! {
+                    r#"
+                    #include <QtGlobal>
+
+                    #if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+                    #include <QtCore/QDir>
+                    #else
+                    #include <QtCore/QtResource>
+                    #endif
+
+                    void
+                    cxx_qt_init_resources()
+                    {{
+                    {init_resources}
+                    }}
+                    "#
+                },
+            ));
+        }
+    }
+
+    fn build_inits_for_qml_plugins(&mut self) {
+        // Build the QML files
+        let import_plugins = self
+            .qml_modules
+            .iter()
+            .map(|qml_module| {
+                // The name is the URI with dots replaced by underscores
+                // and add _plugin to match the CLASS_NAME set in QtBuild::register_qml_module
+                //
+                // https://doc.qt.io/qt-6/qqmlengineextensionplugin.html#Q_IMPORT_QML_PLUGIN
+                let plugin_name = qml_module.uri.replace('.', "_");
+                format!("  Q_IMPORT_PLUGIN({plugin_name}_plugin);")
+            })
+            .collect::<Vec<String>>()
+            .join("\n");
+        if !import_plugins.is_empty() {
+            self.inits.push(CxxQtBuilderInit::new(
+                "cxx_qt_import_plugins",
+                &formatdoc! {
+                    r#"
+                    #include <QtQml/QQmlEngineExtensionPlugin>
+
+                    void
+                    cxx_qt_import_plugins()
+                    {{
+                    {import_plugins}
+                    }}
+                    "#
+                },
+            ));
+        }
+    }
+
+    fn write_init_method(&self) -> String {
+        // Build any initialisers from builder opts
+        let mut call_methods = vec![];
+        let mut sources = vec![];
+
+        for init in &self.inits {
+            call_methods.push(format!("{}();", init.method_name));
+            sources.push(init.source.as_str());
+        }
+
+        let call_methods = call_methods.join("\n");
+        let sources = sources.join("\n");
+
+        let header = formatdoc! {
+            r#"
+            #pragma once
+
+            extern "C" void __cxx_qt_init();
+
+            namespace cxx_qt {{
+            void
+            init();
+            }}
+            "#
+        };
+        let source = formatdoc! {
+            r#"
+            #include <cxx-qt/init.h>
+
+            void
+            __cxx_qt_init()
```

Comment:
When building with `cargo test` we get multiple definitions of some things.

Note we do not see the QML plugins though

```
  = note: /usr/bin/ld.gold: error: /tmp/rustchKpRbk/libqt-static-initializers.a(3e4a4eaacdc769a0-init.o): multiple definition of 'cxx_qt_qt5_compat()'
          /usr/bin/ld.gold: /home/runner/cxx-qt/build/./cargo/build/release/build/qml_minimal-e896cbf14dffc047/out/libqt-static-initializers.a(4c23f01c502e2d66-init.o): previous definition here
          /usr/bin/ld.gold: error: /tmp/rustchKpRbk/libqt-static-initializers.a(3e4a4eaacdc769a0-init.o): multiple definition of 'cxx_qt_lib_core_init()'
          /usr/bin/ld.gold: /home/runner/cxx-qt/build/./cargo/build/release/build/qml_minimal-e896cbf14dffc047/out/libqt-static-initializers.a(4c23f01c502e2d66-init.o): previous definition here
          /usr/bin/ld.gold: error: /tmp/rustchKpRbk/libqt-static-initializers.a(3e4a4eaacdc769a0-init.o): multiple definition of 'cxx_qt_lib_gui_init()'
          /usr/bin/ld.gold: /home/runner/cxx-qt/build/./cargo/build/release/build/qml_minimal-e896cbf14dffc047/out/libqt-static-initializers.a(4c23f01c502e2d66-init.o): previous definition here
          /usr/bin/ld.gold: error: /tmp/rustchKpRbk/libqt-static-initializers.a(3e4a4eaacdc769a0-init.o): multiple definition of 'cxx_qt::init()'
          /usr/bin/ld.gold: /home/runner/cxx-qt/build/./cargo/build/release/build/qml_minimal-e896cbf14dffc047/out/libqt-static-initializers.a(4c23f01c502e2d66-init.o): previous definition here
          /usr/bin/ld.gold: error: /tmp/rustchKpRbk/libqt-static-initializers.a(3e4a4eaacdc769a0-init.o): multiple definition of '__cxx_qt_init'
          /usr/bin/ld.gold: /home/runner/cxx-qt/build/./cargo/build/release/build/qml_minimal-e896cbf14dffc047/out/libqt-static-initializers.a(4c23f01c502e2d66-init.o): previous definition here
          collect2: error: ld returned 1 exit status
```

Reply:
This could be due to them being in both cxx-qt-lib and the target like qml-minimal as they both build using CxxQtBuilder?